staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -91,6 +92,12 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 			KeepAlive: 30 * time.Second,
 		}).DialContext
 	}
+
+	// If we use a cert callback, we need to handle certificate rotation properly
+	if config.HasCertCallback() {
+		dial = certRotatingDialer(tlsConfig, dial, wait.NeverStop)
```

Comment:
Thanks for making this very obvious.  We can now have a discussion about how many of these we think it's ok to reasonable create without closing until the process ends.

@liggitt probably has the most opinions.  I remember him being heavy in this file.

Reply:
paging in context... what's the issue here?

Reply:
leaking go funcs every time a new transport is created.